src/lib.rs

```diff
@@ -50,46 +50,55 @@ pub fn derive_deref_mut(input: TokenStream) -> TokenStream {
     let trait_name = if mutable { "DerefMut" } else { "Deref" };
     let fields = match item.data {
-        syn::Data::Struct(ref body) => body.fields.iter().collect::<Vec<&syn::Field>>(),
+        syn::Data::Struct(ref body) =>
+            body.fields.iter()
+                .filter(|field| {
+                    if let Type::Path(TypePath { path: Path { segments, .. }, .. }) = &field.ty {
+                        if segments
+                            .last()
+                            .expect("Expected path to have at least one segment")
+                            .ident == "PhantomData"
+                        {
+                            false
+                        } else {
+                            true
+                        }
+                    } else {
+                        true
+                    }
+                })
+                .collect::<Vec<&syn::Field>>(),
         _ => panic!("#[derive({})] can only be used on structs", trait_name),
     };
 
-    let field_ty = match fields.len() {
-        1 => Some(fields[0].ty.clone()),
-        2 => {
-            if let Type::Path(TypePath { path: Path { segments, .. }, .. }) = &fields[1].ty {
-                if segments
-                    .last()
-                    .expect("Expected path to have at least one segment")
-                    .value()
-                    .ident == "PhantomData"
-                {
-                    Some(fields[0].ty.clone())
-                } else {
-                    None
+    let target_field = match fields.len() {
+        1 => fields[0],
+        _ => {
+            // Look for attribute
+            let mut targets = Vec::new();
+            for field in fields.iter() {
+                if let Some(_) = field.attrs.iter().find(|attr| attr.path.is_ident("deref_target")) {
+                    targets.push(field);
                 }
-            } else {
-                None
             }
-        },
-        _ => None,
+            match targets.len() {
+                0 => panic!("#[derive({})]: since there is more than one field, #[deref_target] is needed on a field", trait_name),
+                1 => targets[0],
+                _ => panic!("#[derive({})]: only one #[deref_target] allowed", trait_name),
+            }
+
+        }
     };
-    let field_ty = field_ty
-        .unwrap_or_else(|| {
-            panic!(
-                "#[derive({})] can only be used on structs with one field, \
-                 and optionally a second `PhantomData` field.",
-                 trait_name,
-            )
-        });
 
-    let field_name = match fields[0].ident {
+
+
+    let field_name = match target_field.ident {
         Some(ref ident) => quote!(#ident),
         None => quote!(0),
     };
 
-    match (field_ty, mutable) {
-        (syn::Type::Reference(syn::TypeReference { elem, .. }), _) => (*elem.clone(), quote!(self.#field_name)),
+    match (target_field.ty.clone(), mutable) {
```

Comment:
I'm not clear on why this change is needed

Reply:
I think it is, because with this change I needed both the ident and the type, so `target_field` is the field not just the type. clone() is needed because otherwise it moves out of `target_field`